core/modules/node/node.module

```diff
@@ -2370,14 +2370,30 @@ function node_page_view(Node $node) {
 function node_update_index() {
   $limit = (int)config_get('search.settings', 'search_cron_limit');
 
-  $result = db_query_range("SELECT n.nid FROM {node} n LEFT JOIN {search_dataset} d ON d.type = 'node' AND d.sid = n.nid WHERE d.sid IS NULL OR d.reindex <> 0 ORDER BY d.reindex ASC, n.nid ASC", 0, $limit, array(), array('target' => 'slave'));
-  $nids = $result->fetchCol();
-  if (!$nids) {
-    return;
-  }
+  // Index selected content types.
+  $content_types = config_get('search.settings', 'content_types');
```

Comment:
We have another `config_get()` above for `search_cron_limit`, so perhaps call a `$config = config('search.settings');` and then use `$config->get()`s.